aggregator/mock/aggregator.go

```diff
@@ -107,3 +112,30 @@ func (agg *mockAggregator) Snapshot() SnapshotResult {
 
 	return result
 }
+
+func cloneMetric(m unaggregated.MetricUnion) unaggregated.MetricUnion {
+	mu := m
+	if !m.OwnsID {
```

Comment:
nit: Shouldn't we copy the `ID` here as well since we're cloning the metric to ensure it can't be mutated externally? Probably a moot point since we don't intend to modify ID's and this is just a mock but figured I would point it out.